app/views/agents/agent_views/user_location_agent/_show.html.erb

```diff
@@ -17,9 +17,41 @@
     };
 
     var map = new google.maps.Map(document.getElementById("map_canvas"), mapOptions);
+    var circles = [];
+    var points = [];
     <% events.each do |event| %>
-    map_marker(map, <%= Utils.jsonify(event.location) %>);
+    var loc = <%= Utils.jsonify(event.location) %>;
+    if (loc.radius > 1) {
+        circles.push(map_marker(map, loc));
+    }
+    delete loc.radius;
+    points.push(map_marker(map, loc));
     <% end %>
+
+    function Visibility(group, map) {
```

Comment:
Usually capitalized names are reserved for classes.  I'd suggest `setVisibility` or something similar.